packages/prosemirror-lwdita/src/github-integration/request.ts

```diff
@@ -168,62 +157,41 @@ export function processRequest(): undefined | URLParams {
-        // Store the parameters in localStorage for reading the values after the OAuth flow
-        // TODO: After a successful GitHub Authentication, read the user parameters from localStorage and clear it afterwards
-        localStorage.setItem('userParams', JSON.stringify(parameters));
-
-        // Redirect to GitHub OAuth page
-        redirectToGitHubOAuth();
-      } else if (typeof parameters === 'object' && parameters.some(param => isOAuthCodeParam(param.key))) {
-        //TODO(YB): These Params should passed with the OAuth redirect URL
-        // get the stored parameters from localStorage
-        const storedParams = localStorage.getItem('userParams');
-        if (!storedParams) {
-          showErrorPage();
-          return undefined;
-        }
-        // in case of an error, the user did not authenticate the app
-        const errorParam = parameters.find(param => param.key === 'error');
-        if(errorParam) {
-          //TODO(YB): redirect to petal error page
-          // Petal error page should have a button to redirect to the referer
-          return undefined;
-        }
-
-        // exchange the code for an access token
-        const codeParam = parameters.find(param => param.key === 'code');
-        if (!codeParam) return undefined; // I don't understand why this is necessary as the previous if statement should have caught this
-          const code = codeParam.value;
-          exchangeOAuthCodeForAccessToken(code).then(token => {
-          localStorage.setItem('token', token);
-        }).catch(e => {
-          console.error(e);
-        });
-
-        // Show a success notification
-        showNotification("authenticated");
-
+      } else if (typeof parameters === 'object') {
         const returnParams: URLParams = {};
-
-        // Add the stored parameters to the returnParams object
-        if (storedParams) {
-          const stored = JSON.parse(storedParams);
-          for (const param of stored) {
-            returnParams[param.key] = param.value;
-          }
+        for (const param of parameters) {
+          returnParams[param.key] = param.value;
         }
+        if (!parameters.some(param => isOAuthCodeParam(param.key))) {
+
+          // Redirect to GitHub OAuth page
+          redirectToGitHubOAuth(returnParams);
+        } else if (parameters.some(param => isOAuthCodeParam(param.key))) {
+          // in case of an error, the user did not authenticate the app
+          const errorParam = parameters.find(param => param.key === 'error');
+          if(errorParam) {
+            showErrorPage();
+          }
 
-        // return the stored parameters and the new parameters from the URL
-        return returnParams;
+          exchangeOAuthCodeForAccessToken(returnParams.code).then(token => {
+            localStorage.setItem('token', token);
+          }).catch(e => {
+            console.error(e);
```

Comment:
We shouldn't be sending errors to the console still - we should be notifying the user when there is a problem

Reply:
@adamretter Agree, but we don't have an error handling mechanize for that yet, In this case we need to make the user aware he need to authorize Petal again.

Reply:
@marmoure Could we not call `showNotification`?

Reply:
That won't be appropriate as the user won't be able to take any actions.
It will be more appropriate to send the user to an error page.